### PR TITLE
docs: Update Starknet book ToC

### PR DESCRIPTION
**Pull request by microbecode**
Update outdated information about Starknet book

### Description of the Changes

The Table of Contents had outdated information. Update it to reflect the contents of the book.

### PR Preview URL

Update to content at https://docs.starknet.io/tools/starknet-book/ 

### Check List

- [x] Changes have been done against main branch, and PR does not conflict
- [x] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/1297)

### CODE REVIEW

File: components/Starknet/modules/tools/pages/starknet-book.adoc [hunk 1]
```diff
@@ -14,27 +14,23 @@ The Starknet Book caters to various objectives and interests. Mix and match thes
...
+Learn how Starknet works under the hood. Learn how sequencer, prover and nodes interact.
 
 | chapter 4 
-| *Node Operation*
+| *Account abstraction*
 
-Practical insights into operating nodes on the Starknet network.
+Account abstraction is one of the main usability features of Starknet. Learn how this works and how you can utilize it.
 
 | Chapter 5
```
Comment (stoobie) [inline components/Starknet/modules/tools/pages/starknet-book.adoc:32]:
```suggestion

```

File: components/Starknet/modules/tools/pages/starknet-book.adoc [hunk 2]
```diff
@@ -14,27 +14,23 @@ The Starknet Book caters to various objectives and interests. Mix and match thes
...
 | Chapter 5 
-| *Security Analysis*
+| *STARKs*
 
-Comprehensive analysis of security considerations within the Starknet ecosystem.
-| Chapter 6 
-| *Underlying Architecture and Cryptography*
-
-Delve into the foundational architecture and cryptographic principles of Starknet.
+Learn how the proving system STARK works.
```
Comment (stoobie) [inline components/Starknet/modules/tools/pages/starknet-book.adoc:35]:
```suggestion

```

**Review by stoobie — APPROVED** [review]
LGTM

**Comment by stoobie** [comment]
@microbecode Thank you for the contribution! I'm merging this after making one small change, but the Starknet Book is now deprecated, and we'll remove this page shortly.

**Comment by microbecode** [comment]
Oh I didn't know it's deprecated :) Ok, thanks!

**Pull request merged**